### ISSUE TITLE
ext_proc crash when using grpc initial_metadata and REQ/RESP headers on 1.18+

### ISSUE THREAD

**Issue by pgeler** [envoyproxy/envoy#18099]
Issue was originally submitted to [envoy-security](https://groups.google.com/d/msgid/envoy-security/c1a201e7-ba08-4c55-85b6-6ca5742b4713n%40googlegroups.com?utm_medium=email&utm_source=footer), but rejected from there confirming that it was reproducible, asking to create a ticket in github:

dynamic functions like %REQ()% or %RESP()% causing crash for v1.18/v1.19. On v1.17 passing without crash.
config snippet, does not require working service, was tested with simple http:
```
          http_filters:
          - name: extension-matcher
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.ext_proc.v3alpha.ExternalProcessor
              failure_mode_allow: true
              message_timeout: 5s
              processing_mode:
                request_header_mode: "SEND"
                response_header_mode: "SEND"
                request_body_mode: "NONE"
                response_body_mode: "NONE"
                request_trailer_mode: "SKIP"
                response_trailer_mode: "SKIP"
              grpc_service:
                envoy_grpc:
                  cluster_name: origin
                initial_metadata:
                - key: "x-request-id"
                  value: "%REQ(x-request-id)%"
          - name: envoy.filters.http.router

**Comment by alyssawilk** [comment]
cc @gbrail @snowp

**Comment by gbrail** [comment]
It looks like the problem is that we don't pass the "StreamInfo" to the gRPC stream when starting to talk to the ext_proc server. That's pretty easy to fix, and harder to write an integration test for but I'll work on it. Thanks!

**Comment by pgeler** [comment]
issue solved in #18190
